services/seller.js

```diff
@@ -8,46 +8,28 @@ const models = require('../models');
 module.exports = {
 
   createSeller: (seller) => {
-    return new Promise((resolve) => {
-      bcrypt.hash(seller.password, config.bcryptSaltRounds).then((hash) => {
-        seller.password = hash;
-
-        // insert seller info to db
-        models.seller.create(seller).then(seller => {
-          resolve(seller.dataValues);
-        });
-      });
-    });
+    const hash = bcrypt.hashSync(seller.password, config.bcryptSaltRounds);
+    seller.password = hash;
+
+    return models.seller.create(seller).then(seller => seller.dataValues);
   },
 
-  updateSeller: (params) => {
-    return new Promise((resolve, reject) => {
-      models.seller.findOne({
-        where: {
-          id: params.id
-        }
-      }).then(seller => {
-        if (seller) {
-          // update password
-          if (params.password) {
-            bcrypt.hash(params.password, config.bcryptSaltRounds)
-              .then((hash) => {
-                params.password = hash;
-                seller.updateAttributes(params)
-                  .then(seller => {
-                    resolve(seller.dataValues);
-                  });
-              });
-          } else {
-            seller.updateAttributes(params)
-              .then(seller => {
-                resolve(seller.dataValues);
-              });
-          }
-        } else {
-          reject('No such seller Exist');
-        }
-      });
+  updateSeller: params => {
+    return models.seller.findOne({
+      where: {
+        id: params.id
+      }
+    }).then(seller => {
+      if (!seller) {
+        throw ('No such seller Exist');
+      }
+
+      // update password
+      if (params.password) {
+        params.password = bcrypt.hashSync(params.password, config.bcryptSaltRounds);
```

Comment:
I think we should go for async bcrypt functionality for our code to be future proof. 
Regarding cleaner code - we can use `async await` for promises
```
return models.seller.findOne({
      where: {
        id: params.id
      }
    }).then(async seller => {
      if (!seller) {
        throw ('No such seller Exist');
      }
      // update password
      if (params.password) {
        params.password = await bcrypt.hash(params.password, config.bcryptSaltRounds);
      }
      return seller.updateAttributes(params).then(seller => seller.dataValues);
    });
```

Reply:
Sure, this looks good.